dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/impl/CachedCompositeProperty.java

```diff
@@ -139,14 +157,14 @@ private T firstNonNull(T first, T second) {
      * @return
      * @throws Exception
      */
-    protected abstract T resolveCurrent() throws Exception;
+    protected abstract CachedCompositeValue<T> resolveCurrent() throws Exception;
 
     /** {@inheritDoc} */
     @Override
     public int hashCode() {
```

Comment:
Needs a javadoc comment as so odd and looks like a missing bit (the value) but probably as intended (will be odd though if used as a key in a map? as all Strings will have the same hash).